src/libraries/ref.proj

```diff
@@ -0,0 +1,24 @@
+<Project Sdk="Microsoft.Build.Traversal">
+
+  <PropertyGroup>
+    <TraversalGlobalProperties>BuildAllProjects=true</TraversalGlobalProperties>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <GeneratedShimProject Include="shims\generated\*.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="$(MSBuildThisFileDirectory)*\ref\*.*proj" Exclude="@(ProjectExclusions)" />
+    <!-- Restore only and build later. -->
+    <ProjectReference Include="@(GeneratedShimProject)" Condition="'$(MSBuildRestoreSessionId)' != ''" />
```

Comment:
Is MSBuildRestoreSessionId the way to identify if we're in the restore phase? That seems more like implementation detail, but I don't know.

Reply:
Yes: https://github.com/microsoft/msbuild/blob/b111470ae61eba02c6102374c2b7d62aebe45f5b/src/MSBuild/XMake.cs#L1425.

Reply:
Oh, I see you pass it down as a global property, was that just for this, or MSBuild uses that as well?

Reply:
and https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj#L241

Reply:
Cool, didn't know that.